File: BeerViewer/Models/Raw/kcsapi_port.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BeerViewer.Models.Raw
{
	public class kcsapi_port
	{
		public kcsapi_material[] api_material { get; set; }
		public kcsapi_deck[] api_deck_port { get; set; }
		public kcsapi_ndock[] api_ndock { get; set; }
		public kcsapi_ship2[] api_ship { get; set; }
		public kcsapi_basic api_basic { get; set; }
		public int api_combined_flag { get; set; }
		//public Api_Log[] api_log { get; set; }
	}
}

File: WindowFrame/source/System/Interfaces/kcsapi_port.ts
import { kcsapi_material } from "./kcsapi_material";
import { kcsapi_deck } from "./kcsapi_deck";
import { kcsapi_ndock } from "./kcsapi_dock";
import { kcsapi_basic } from "./kcsapi_basic";
import { kcsapi_ship2 } from "./kcsapi_ship";

export interface kcsapi_port {
	api_material: kcsapi_material[];
	api_deck_port: kcsapi_deck[];
	api_ndock: kcsapi_ndock[];
	api_ship: kcsapi_ship2[];
	api_basic: kcsapi_basic;
	api_combined_flag: number;
}
